lib/routes/keylol/index.ts

```diff
@@ -1,36 +1,61 @@
 import { Route } from '@/types';
-import { getSubPath } from '@/utils/common-utils';
 import cache from '@/utils/cache';
 import got from '@/utils/got';
 import { load } from 'cheerio';
 import timezone from '@/utils/timezone';
 import { parseDate } from '@/utils/parse-date';
+import parser from '@/utils/rss-parser';
+import queryString from 'query-string';
+
+const threadIdRegex = /(\d+)-\d+-\d+/;
 
 export const route: Route = {
-    path: '*',
-    name: 'Unknown',
-    maintainers: [],
+    path: '/:path',
+    name: '论坛',
+    parameters: { path: '路径，默认为热点聚焦' },
+    categories: ['game', 'bbs'],
+    example: '/keylol/f161-1',
+    radar: [
+        {
+            source: ['keylol.com/:path'],
+            target: (params, url) => url.replaceAll('forum.php?', ''),
+        },
+    ],
+    maintainers: ['nczitzk', 'kennyfong19931'],
     handler,
+    description: `:::tip
+  若订阅 [热点聚焦](https://keylol.com/f161-1)，网址为 \`https://keylol.com/f161-1\`。截取 \`https://keylol.com/\` 到末尾的部分 \`f161-1\` 作为参数，此时路由为 [\`/keylol/f161-1\`](https://rsshub.app/keylol/f161-1)。
+  若订阅子分类 [试玩免费 - 热点聚焦](https://keylol.com/forum.php?mod=forumdisplay&fid=161&filter=typeid&typeid=459)，网址为 \`https://keylol.com/forum.php?mod=forumdisplay&fid=161&filter=typeid&typeid=459\`。提取\`fid\`及\`typeid\` 作为参数，此时路由为 [\`/keylol/fid=161&typeid=459\`](https://rsshub.app/keylol/fid=161&typeid=459)。注意不要包括\`filter\`，会调用[全局的内容过滤](https://docs.rsshub.app/guide/parameters#filtering)。
+  :::`,
 };
 
 async function handler(ctx) {
-    let thePath = getSubPath(ctx).replace(/^\//, '');
-
-    if (/^f\d+-\d+/.test(thePath)) {
-        thePath = `fid=${thePath.match(/^f(\d+)-\d+/)[1]}`;
+    let queryParams = {};
+    const path = ctx.req.param('path');
+    if (/^f\d+-\d+/.test(path)) {
+        queryParams.fid = path.match(/^f(\d+)-\d+/)[1];
+    } else {
+        queryParams = queryString.parse(path);
     }
+    queryParams.mod = 'forumdisplay';
+    queryParams.orderby = 'dateline';
+    queryParams.filter = 'author';
 
-    const limit = ctx.req.query('limit') ? Number.parseInt(ctx.req.query('limit'), 10) : 30;
+    // get real author name from official rss feed
+    const feed = await parser.parseURL(`https://keylol.com/forum.php?mod=rss&fid=${queryParams.fid}&auth=0`);
+    const authorNameMap = feed.items.map((item) => ({
+        threadId: item.link.match(threadIdRegex)[1],
+        author: item.author,
+    }));
 
     const rootUrl = 'https://keylol.com';
-    const currentUrl = new URL(`forum.php?mod=forumdisplay&${thePath.replaceAll(/mod=\w+&/g, '')}`, rootUrl).href;
+    const currentUrl = queryString.stringifyUrl({ url: `${rootUrl}/forum.php`, query: queryParams });
 
     const { data: response } = await got(currentUrl);
 
     const $ = load(response);
 
-    let items = $('a.xst')
-        .slice(0, limit)
```

Comment:
Why remove the [commmon parameter limit](https://docs.rsshub.app/guide/parameters#limit-entries)?

Reply:
First thought it was useless as a limit is a common param handled by middleware.

Added back to reduce rest calls, especially for the post with a table of contents that I have added support to.